WooCommerce/Classes/ViewModels/Order Details/OrderDetailsViewModel.swift

```diff
@@ -409,6 +409,19 @@ extension OrderDetailsViewModel {
         ServiceLocator.stores.dispatch(action)
     }
 
+    func checkCardPaymentEligibility(onCompletion: (() -> Void)? = nil) {
+        // For now, this defaults to the feature flag,
+        // combined with the order payment method being cash on delivery
+        // We also need to take into account wheter the store is enrolled into WCPay.
+        // This is synchronous for now, but it will be async in the nearly future:
+        // https://github.com/woocommerce/woocommerce-ios/issues/3828
+        // We need to add a unit test for this, after the logic is settled
+        dataSource.isEligibleForCardPresentPayment = ServiceLocator.featureFlagService.isFeatureFlagEnabled(.cardPresentPayments) &&
```

Comment:
We should only allow payment on orders with status "pending payment" or "on hold"

We should never allow payment on orders with status "cancelled", "processing," "refunded" or "completed" 

We may want to allow payment on orders with status "failed" or "authentication required" - this could be useful if the card they attempted first was declined and they wish to try again with a card present payment. Or, if you prefer, it would also make sense to disallow these two as well for now and open an issue to settle this later.

Reply:
I think it would be easier if we go with pending payment and on hold for now, until we have more clarity.
Added an issue: #3857

Reply:
I am curious., why not on orders with status "processing"? If I order something on a store (web), that and select "cash on delivery" that'd be the status the order is in, as far as I can tell.

Reply:
![surprised-pikachu](https://user-images.githubusercontent.com/1595739/112215887-2124ea00-8bde-11eb-87a3-50fa6f553560.png)

I'm surprised! Our [docs](https://docs.woocommerce.com/document/managing-orders/) say:

> Processing — Payment received (paid) and stock has been reduced; order is awaiting fulfillment.

But sure enough, `cod` surprisingly leaves the order in `processing` - this looks like an area of some contention and I think there is a bug in core: https://github.com/woocommerce/woocommerce/pull/15268/files - I'll investigate and follow up with core

In the meantime, yeah, I'd allow `processing` for `cod` too

Reply:
I'll push a quick update for that

Reply:
https://github.com/woocommerce/woocommerce/issues/29471